### ISSUE TITLE
Error install app The redirect_uri is missing

### ISSUE THREAD

**Issue by khetesh** [phpish/shopify_app-skeleton#32]
Hello ,
I seem to be getting an error now when using the SDK: 
Like this error first image :-
1. 
![shopifyerror](https://user-images.githubusercontent.com/19745787/27826736-cf30df78-60d1-11e7-8720-b3926a8e918d.PNG)

my callback URL is install.php as explained in the readme (Image 1)

shopify partners account s config. image like this.

![shopifyerror2](https://user-images.githubusercontent.com/19745787/27826789-0fb9ac0a-60d2-11e7-899b-b6dfca16de63.PNG)

**Comment by myjanky** [comment]
What exactly is the error you are receiving?

**Comment by khetesh** [comment]
Invalid Request! Request or redirect did not come from Shopify. 
 oauth.php  this line
shopify\is_valid_request($_GET, SHOPIFY_APP_SHARED_SECRET) or die('Invalid Request! Request or redirect did not come from Shopify');

**Comment by myjanky** [comment]
seems like you are in a redirect loop. Also did you change the md5 to hmac?

**Comment by khetesh** [comment]
Thank you for your explanation. I finally go this to work.

**Comment by khetesh** [comment]
How to custom setting implemented from app to shopify website explanation about linking?

Hello sir,


Now I am stuck for next step. I don't understand how can I linked the custom setting implemented on my app to shopify frontend. How can I linked the app to shopify website.
How can I implement on shopify website for customer to customize any product and where I save all the customized data and how can I linked that data with shopify website.

I am able to make standalone app with php but don't understand how its data can be linked to shopify website. How my server app database can be used in shopify website.

My server SSL supported.

My app url  :- https://apps.shopify.com/come-hello

I want Real-time notification of recent sales to increase customer buying products on website store live notification.

How can I do this? How do I actually use this app to interact (integrate) on my web site frontend? But I want like this app build using php SDK 

For example like this apps url :- https://apps.shopify.com/sales-pop  this is possible in PHP.

I am new here. Any help is very much appreciated.                            

 

Thanks

**Comment by myjanky** [comment]
@khetesh Yes I can help you. Let's break up your request into smaller tasks and then you can put it all together. 
In order to get your app to talk to the front end of your shop you will need to write some Javascript to talk to your server. I usually use an iframe and a JS GET ajax to load the content into the liquid page (you can also send liquid from your server to have it render on your shop but you will have to edit the header to allow liquid).

I can show you how to do this. Are you comfortable with JS?

**Comment by khetesh** [comment]
Yes,i comfortable with JS. How do it use in js with my app?

**Comment by myjanky** [comment]
https://github.com/phpish/shopify_app-skeleton/issues/21
see this and let me know if you have any questions.

**Comment by ramakrishnaj** [comment]
seems like you are in a redirect loop. Also did you change the md5 to hmac?

hello sir

I am getting same this issue . where we need to change  md5 to hmac?
